client/src/com/mirth/connect/client/ui/actions/PasteAction.java

```diff
@@ -35,8 +36,14 @@ public void actionPerformed(ActionEvent e) {
     public boolean isEnabled() {
         if (comp.isVisible() && comp.isEditable() && comp.isEnabled()) {
             try {
-                Transferable contents = Toolkit.getDefaultToolkit().getSystemClipboard().getContents(this);
-                return contents.isDataFlavorSupported(DataFlavor.stringFlavor);
+                Clipboard clipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
+                if(null != clipboard) {
+                    Transferrable contents = clipboard.getContents(this);
+                    if(null != contents) {
+                        return contents.isDataFlavorSupported(DataFlavor.stringFlavor);
+                    }
+                }
+                return false;
             } catch (IllegalStateException e) {

```

Comment:
When would an IllegalStateException be thrown?

Should this exception be at least logged and not hidden?

Reply:
@jonbartels Apparently the IllegalStateException can be thrown on the call to ```.getContents(this)``` if the clipboard is unavailable. The example that the documentation provided is where on some platforms, the system clipboard is unavailable if it's being accessed by another application.

I don't think we need to log this exception since a user won't be able to do anything with or within Mirth Connect to resolve it.